### PR TITLE
Validate input numpy

### PR DESCRIPTION
**Pull request by facusapienza21**
Solution to Issue #82 

I added numpy scalars as available inputs and now we can use np.arrays as inputs for region_extent too.

### CODE REVIEW

File: icepyx/core/validate_inputs.py [hunk 1]
```diff
@@ -54,7 +59,11 @@ def spatial(spatial_extent):
             extent_type = "bounding_box"
 
         # user-entered polygon as list of lon, lat coordinate pairs
-        elif all(type(i) in [list, tuple] for i in spatial_extent):
+        elif all(type(i) in [list, tuple, np.ndarray] for i in spatial_extent) and all( 
+            all( isinstance(i[j], scalar_types) for j in range(len(i)) ) for i in spatial_extent
+        ):
+            if any( len(i) != 2 for i in spatial_extent):
+                raise ValueError("Each element in spatial_extent should be a list or tuple of lenght 2")
```
Comment (JessicaS11) [inline icepyx/core/validate_inputs.py:66]: typo

**Comment by JessicaS11** [comment]
Thanks @facusapienza21. These changes look great! Do you think you'd be able to add some tests to the test suite to make sure we're testing these additions? It should be fairly straightforward since there are already some tests written for the validation section, but I'd be happy to help if you're new to unit testing.

**Comment by facusapienza21** [comment]
Hi @JessicaS11 Sorry for the late response. Yes! I can do that. Let me add some examples + correct the typo.

**Comment by facusapienza21** [comment]
Hello @JessicaS11, I made a few modifications and I add two tests. Something else you want me do add? I made the push of my modification, but do I have to do a new pull request? Thank you!

**Comment by JessicaS11** [comment]
Thanks @facusapienza21  - this looks great. When you push to the same branch as an open PR, it automatically adds the commits to the PR, so there's no need to submit a new one. I've merged your commits into the development branch!

**Pull request merged**